Forge Standard Library docs (#207)
foundry-rs/book: src/reference/forge-std/std-storage.md

> ## Std-storage

Reviewer: ```suggestion
## Std Storage
```

> To use std-storage, add the following line to your test contract:

Reviewer: ```suggestion
To use Std Storage, add the following line to your test contract:
```

Additionally, we should give a short explainer of what Std Storage is: it's a library to manipulate the storage of contracts.

```solidity
using stdStorage for StdStorage;
```

> Then, access std-storage via the `stdstore` instance.

Reviewer: ```suggestion
Then, access Std Storage via the `stdstore` instance.
```

> ### Functions

Reviewer: Do you need to call all of these functions, or can you chain them as you wish? I.e. is `target(someContract).sig(someFunction.selector).find()` valid? What if I don't call `target` but instead `sig(someSig).find()`? 

If not, we should probably split these into:

- Query functions (target, sig, slot etc.): Limits your search for a storage slot
- Terminator functions (`checked_write`, `find`): Performs the query (either a write or a read)

We should additionally explain if any of these are required

- `target`: Set contract address.

> - `sig`: Set function signature.

Reviewer: ```suggestion
- `sig`: Set function signature to call.
```

> - `with_key`: Set key, if variable is of type `mapping`.

Reviewer: ```suggestion
- `with_key`: Set key, if the slot you are looking for is a mapping.
```

Author: Maybe we should come up with a better explanation for this one and the one below

> - `depth`: Set depth, if variable is of type `struct`.

Reviewer: I assume `depth` here is actually the number of the member variable you are looking up in the struct? E.g. if I have a struct:

```solidity
struct Abc {
  uint256 x; // depth = 0
  uint256 y; // depth = 1
}
```

If so, we should try to convey that in a simple manner, e.g. "Set the index of the struct member if the slot you are looking for is a struct" or something.

> - `checked_write`: Set data to be written.

Reviewer: ```suggestion
- `checked_write`: Set data to be written to the storage slot.
```

Additionally, what happens if the query matches multiple slots? What if it matches no slots?

Author: Help. I need to familiarize myself with Std Storage inner workings more ><

Also, we'll probably need to add a Troubleshooting section to Std Storage (later)

> - `find`: Return slot number.

Reviewer: Same question as above
